### ISSUE TITLE
Character object? (question)

### ISSUE THREAD

**Issue by Splitwirez** [aaartrtrt/rendisco#5]
I've noticed that `IRuntimeEngine` has a few comments which suggest that certain methods "could be enhanced to return Character object"...and yet, despite the changes you've been making as of recently, that hasn't happened. I'm curious...have you decided to not make that change? If so, why is that? :eyes:

(to clarify, I'm not trying to suggest you *should* or *shouldn't*, I'm merely asking out of curiousity)

---

**Full disclosure**: Admittedly, representing characters via a dedicated type intuitively *seems* like a good idea to me. That intuition is wholly unfounded, though, as I have very little experience with Python and even less with Ren'Py or anything like it.

---

Also, speaking of `IRuntimeEngine`...is there a reason why the actual XML documentation is only present on `BaseRuntimeEngine`, and not on `IRuntimeEngine` itself?

**Comment by aaartrtrt** [comment]
It should be! The only reason why it isn't there is that I only really get time to work on this project after work or on the weekends. If possible, the Character object should really be implemented, alongside any dynamically defined resource in RenPy. I'm currently in the spiral of trying to cover ANTLR parsing for RenPy to save me having to rewrite the lexer and parser every time I support a new bit of the language specification. I'll look into setting up a public project so the lines of effort is transparent :^)

**Comment by Splitwirez** [comment]
> It should be! The only reason why it isn't there is that I only really get time to work on this project after work or on the weekends.

@aaartrtrt Right, fair enough - thus is the nature of hobby projects. Can relate :+1: 


> If possible, the Character object should really be implemented, alongside any dynamically defined resource in RenPy.

Hmmm...any idea if you'd be able to accept a PR to take care of that? :thinking: 

---

> I'm currently in the spiral of trying to cover ANTLR parsing for RenPy to save me having to rewrite the lexer and parser every time I support a new bit of the language specification.

![Incomprehensible, have a nice day.](https://github.com/user-attachments/assets/6f0be482-c754-4ba3-b597-60de44cd1fa6)


> I'll look into setting up a public project so the lines of effort is transparent :^)

Nice

**Comment by aaartrtrt** [comment]
> Hmmm...any idea if you'd be able to accept a PR to take care of that? 🤔

I'm going through a refactor to add a Function Call handler that will treat `Character` as a function call per #14. I hope once this stretch of work is done, it's as close as it gets to somewhere I want it to be.
